qiskit_experiments/library/randomized_benchmarking/rb_analysis.py

```diff
@@ -235,4 +230,50 @@ def _extra_database_entry(self, fit_data: curve.FitData) -> List[AnalysisResultD
                                 device_components=[Qubit(i) for i in qubits],
                             )
                         )
+
         return extra_entries
+
+    def _run_analysis(
+        self, experiment_data: ExperimentData
+    ) -> Tuple[List[AnalysisResultData], List["pyplot.Figure"]]:
+
+        if self.options.gate_error_ratio is None:
+            gate_error_ratio = experiment_data.metadata.get("gate_error_ratio", None)
+
+            try:
+                gate_error_ratio = dict(gate_error_ratio)
+            except TypeError:
+                pass
+
+            if gate_error_ratio is None:
+                # For backward compatibility when loading old experiment data.
+                # This could return errorneous error ratio.
+                # Deprecation warning is triggered on RBUtils.
+                gate_error_ratio = RBUtils.get_error_dict_from_backend(
+                    backend=experiment_data.backend,
+                    qubits=experiment_data.metadata["physical_qubits"],
+                )
+
+            self.set_options(gate_error_ratio=gate_error_ratio)
+
+        if self.options.gate_counts_per_clifford is None:
+            gpc = experiment_data.metadata.get("gate_counts_per_clifford", None)
+
+            try:
```

Comment:
Same here. Also what is the case for needing to explicitly convert to a dict?

Reply:
This is needed to be a dictionary but for JSON serialization this is stored as tuple, i.e. `((key1, val1), (key2, val2), ...)`. Note that `key*` is `(qubit index: int, gate: str)` which is not JSON serializable dictionary, i.e. key should be string.